lib/src/where_type.dart

```diff
@@ -21,25 +21,22 @@ class _WhereType<R> extends StreamTransformerBase<Null, R> {
     StreamSubscription<Object> subscription;
     controller.onListen = () {
       if (subscription != null) return;
-      var valuesDone = false;
       subscription = values.listen(
           (value) {
             if (value is R) controller.add(value);
           },
           onError: controller.addError,
           onDone: () {
-            valuesDone = true;
+            subscription = null;
             controller.close();
           });
       if (!values.isBroadcast) {
         controller.onPause = subscription.pause;
         controller.onResume = subscription.resume;
       }
       controller.onCancel = () {
-        var toCancel = subscription;
+        subscription?.cancel();
```

Comment:
this used to return this value, not sure if it matters or not

Reply:
Hmm, I'm not sure either. There was some weirdness with an analyzer hint around this at one point that could have been why...